lib/utils/time.go

```diff
@@ -24,21 +24,6 @@ import (
 	"github.com/jonboulle/clockwork"
 )
 
-// MinTTL finds min non 0 TTL duration,
```

Comment:
Note: technically, the `min` builtin has different semantics than what's here, but looking at the call sites we should be OK.

This func is a mess anyway - it sometimes returns the min but other times returns the max, and its godoc mentions behavior that doesn't exist (when a == b)..

Reply:
Well, bummer. Turns out we rely on this cursed behavior. I will restore this but at least name it better.